documentation/dsls/DSL-Ash.Domain.md

```diff
@@ -126,6 +126,7 @@ define :get_user_by_id, action: :get_by_id, args: [:id], get?: true
 | [`get?`](#resources-resource-define-get?){: #resources-resource-define-get? } | `boolean` | `false` | Expects to only receive a single result from a read action or a bulk update/destroy, and returns a single result instead of a list. Sets `require_reference?` to false automatically. |
 | [`get_by`](#resources-resource-define-get_by){: #resources-resource-define-get_by } | `atom \| list(atom)` |  | Takes a list of fields and adds those fields as arguments, which will then be used to filter. Sets `get?` to true and `require_reference?` to false automatically. Adds filters for read, update and destroy actions, replacing the `record` first argument. |
 | [`get_by_identity`](#resources-resource-define-get_by_identity){: #resources-resource-define-get_by_identity } | `atom` |  | Takes an identity, gets its field list, and performs the same logic as `get_by` with those fields. Adds filters for read, update and destroy actions, replacing the `record` first argument. |
+| [`default_options`](#resources-resource-define-default_options){: #resources-resource-define-default_options } | `any` | `[]` | Default options used when none are supplied. These can override domain or action defaults. |
```

Comment:
"When none are supplied" isn't quite right here is it? It merges options, even when some are supplied right?

We should also likely merge at one level deeper for `bulk_options`.

Reply:
I should have just slept on it, lol. The last commit where I cleaned up my own mistakes implied I was tired… but I was in the zone. 

I’ll fix the docs a little later. I don’t actually understand the comment about `bulk_options`.

Reply:
There is a nested set of options called `bulk_options` that is accepted by code interfaces. If the caller specified `bulk_options` as well as the defaults we should merge the two values instead of overwriting.

Reply:
I'm a bit lost on the `bulk_options` comment. So far as I can tell, `:default_options` merges _before_ `:bulk_options`. That seems like the correct behavior.

My next project might be a full on refactor of that module. It'd be a good project to improve my macro skills, but there's a lot of duplication. Searching for `bulk_options` in the code led me to a bit of a... "Hmmm, where do I even start?"

Do you want to merge this as is (I pushed to doc changes) and I can change the way the bulk options resolve in another PR?